pkg/kubectl/cmd/convert/convert.go

```diff
@@ -118,9 +118,11 @@ func (o *ConvertOptions) Complete(f cmdutil.Factory, cmd *cobra.Command) (err er
 	}
 	o.builder = f.NewBuilder
 
-	o.Namespace, _, err = f.ToRawKubeConfigLoader().Namespace()
-	if err != nil {
-		return err
+	if !o.local {
```

Comment:
Although this solves the problem, that's not the right approach. See what I wrote in https://github.com/kubernetes/kubernetes/issues/93188, towards the bottom. You picked the hacky solution which we don't want to happen. The entire sig including the tech leads would like to see a more general solution, rather this one off.